lib/KDateRange/KDateCalendar.vue

```diff
@@ -0,0 +1,475 @@
+        if (result < 1 || result > activeMonthDate) return false;
+        var currDate = this.getDate(key, result);
+        return (
+          this.selectedStartDate &&
+          this.selectedEndDate &&
+          this.selectedStartDate < currDate &&
+          this.selectedEndDate > currDate
+        );
+      },
+      /**
+       * returns true for dates that are not apart of the current month; should be hidden from view
+       */
+      isDateDisabled(weekIndex, dayInWeekIndex, activeMonthDay, activeMonthDate) {
+        const result = this.getDayIndexInMonth(weekIndex, dayInWeekIndex, activeMonthDay);
+        // bound by > 0 and < last day of month
+        return !(result > 0 && result <= activeMonthDate);
+      },
+      /**
+       * returns true for disabled dates before the firstAllowedDate; should be visible but grayed out
+       */
+      isDateDisabledLeft(weekIndex, dayInWeekIndex, activeMonthDay) {
+        const result = this.getDayIndexInMonth(weekIndex, dayInWeekIndex, activeMonthDay);
+        const currDate = new Date(this.activeYearStart, this.activeMonth, result);
+        return currDate < this.firstAllowedDate || currDate > this.lastAllowedDate;
+      },
+      /**
+       * returns true for disabled dates after the lastAllowedDate; should be visible but grayed out
+       */
+      isDateDisabledRight(weekIndex, dayInWeekIndex, activeMonthDay) {
+        const result = this.getDayIndexInMonth(weekIndex, dayInWeekIndex, activeMonthDay);
+        const currDate = new Date(this.activeYearStart, this.activeMonth + 1, result);
+        return currDate < this.firstAllowedDate || currDate > this.lastAllowedDate;
+      },
+      /**
+       * return true if date is last day of month for css border rounding
+       */
+      isLastDay(weekIndex, dayInWeekIndex, key, nextActiveMonthDay, nextActiveMonthDate) {
+        const result = this.getDayCell(
+          weekIndex,
+          dayInWeekIndex,
+          nextActiveMonthDay,
+          nextActiveMonthDate
+        );
+        let lastDay = null;
+        if (key === 'first') {
+          lastDay = new Date(this.activeYearStart, this.activeMonth + 1, 0).getDate();
+        } else {
+          lastDay = new Date(this.activeYearEnd, this.nextActiveMonth + 1, 0).getDate();
+        }
+        return result === lastDay;
+      },
+      goPrevMonth() {
+        const prevMonth = new Date(this.activeYearStart, this.activeMonth, 0);
+        this.activeMonth = prevMonth.getMonth();
+        this.activeYearStart = prevMonth.getFullYear();
+        this.activeYearEnd = prevMonth.getFullYear();
+      },
+      goNextMonth() {
+        const nextMonth = new Date(this.activeYearEnd, this.nextActiveMonth, 2);
+        this.activeMonth = nextMonth.getMonth();
+        this.activeYearStart = nextMonth.getFullYear();
+        this.activeYearEnd = nextMonth.getFullYear();
+      },
+      isValidDate(date) {
+        return new Date(date) !== 'Invalid Date' && !isNaN(new Date(date));
```

Comment:
Is `'Invalid Date'` browser specific? Also, what if they have their browser in a different language?

Reply:
This is a good point, maybe the `!isNaN(new Date(date))` should be enough to validate the `date` argument.